doc/cargo_command_execution.md

```diff
@@ -1,108 +1,114 @@
 # Cargo Command Execution Page
 
-The extension allows a developer to execute any of built-in cargo commands.
+The extension allows a developer to execute any of the inbuilt Cargo commands.
 
 These commands are:
 
-* bench
-* build
-* check
-* clean
-* clippy
-* doc
-* new
-* run
-* test
-* update
+* `bench`
+* `build`
+* `check`
+* `clean`
+* `clippy`
+* `doc`
+* `new`
+* `run`
+* `test`
+* `update`
 
-These commands available through the command palette (CTRL+P).
-
-These commands have prefix `"Cargo: "`.
+These commands are available through the command palette (<kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>) and have the prefix `"Cargo: "`.
 
 ## Execute Command On Save
 
-The extension supports executing a command after saving the document opened in the active text document.
+The extension supports executing one of these commands after saving the active document. 
+
+The `"rust.actionOnSave"` configuration parameter specifies which command to execute.
+
+The possible values are:
+
+* `"build"` - executes `"Cargo: Build"`
+* `"check"` - executes `"Cargo: Check"`
+* `"clippy"` - executes `"Cargo: Clippy"`
+* `"run"` - executes `"Cargo: Run"`
+* `"test"` - executes `"Cargo: Test"`
+* `null` - the extension does nothing (default)
+
+## Finding Out Cargo.toml
 
-The `"rust.actionOnSave"` configuration parameter specifies a command to execute.
+Before executing the command, the extension needs to find out which `Cargo.toml` to use. The extension uses the following algorithm:
 
-The possible values:
+* Try to determine the current working directory from the active text editor
 
-* `"build"` - the extension executes `"Cargo: Build"`
-* `"check"` - the extension executes `"Cargo: Check"`
-* `"clippy"` - the extension executes `"Cargo: Clippy"`
-* `"run"` - the extension executes `"Cargo: Run"`
-* `"test"` - the extension executes `"Cargo: Test"`
-* `null` - the extension does nothing
+  If all of the following conditions are met:
 
-By default, it is `null`.
+  * There is an active text editor
+  * A file opened in the editor is within the workspace (the directory opened in VS Code)
+  * There is a `Cargo.toml` in the same directory as the active file or in any of the parent directories within the workspace
 
-## Current Working Directory Determination
+  Then use the `Cargo.toml` file that was found.
 
-The extension executes a cargo command in some directory. To find out which directory the extension should use, the extension uses the following algorithm:
+* Try using the previous `Cargo.toml` file
+* Try using the `Cargo.toml` from the workspace
 
-* Try making out the current working directory from the active text editor
+If the extension fails to find a `Cargo.toml`, an error message is shown.
```

Comment:
I think I changed it because after the first set of bullet points, you say "Then use the `Cargo.toml` file." suggesting what *will* happen. Therefore, I changed it to reflect, *if* the extension fails, X happens.

Would and will are very similar, it's hard for me to express exactly when and where to use each. I think would is generally used *if* something. Will is more stronger and definitive. They can often be used interchangeably. For example, I cannot express that I will make a commit by saying I would make a commit, it implies: you would, but why not? I do not know Russian, but English has many quirks that simply must be learnt and is not as predictable as languages such as German or Japanese.